src/web/client/common/authenticationProvider.ts

```diff
@@ -161,3 +161,62 @@ export async function npsAuthentication(
 
     return accessToken;
 }
+
+export async function graphClientAuthentication(
+    firstTimeAuth = false
+): Promise<string> {
+    let accessToken = "";
+    try {
+        let session = await vscode.authentication.getSession(
+            PROVIDER_ID,
+            [
+                `Contacts.Read`,
+                `User.ReadBasic.All`,
+            ],
+            { silent: true }
+        );
+
+        if (!session) {
+            session = await vscode.authentication.getSession(
+                PROVIDER_ID,
+                [
+                    `Contacts.Read`,
```

Comment:
save these as constants and consume here - similar to constants we use in dataverseAuthentication call